spanner/spanner_snippets/spanner/pg_spanner_create_database.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spanner
+
+// [START spanner_postgresql_create_database]
+import (
+	"context"
+	"fmt"
+	"io"
+	"regexp"
+
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	adminpb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+)
+
+// pgCreateDatabase shows how to create a Spanner database that uses the
+// PostgreSQL dialect.
+func pgCreateDatabase(ctx context.Context, w io.Writer, db string) error {
+	// db := "projects/my-project/instances/my-instance/databases/my-database"
+	matches := regexp.MustCompile("^(.*)/databases/(.*)$").FindStringSubmatch(db)
+	if matches == nil || len(matches) != 3 {
+		return fmt.Errorf("Invalid database id %s", db)
+	}
+
+	adminClient, err := database.NewDatabaseAdminClient(ctx)
+	if err != nil {
+		return err
+	}
+	defer adminClient.Close()
+
+	opCreate, err := adminClient.CreateDatabase(ctx, &adminpb.CreateDatabaseRequest{
```

Comment:
NIT: Consider removing struct instantiation to outside of the call to `CreateDatabase()`. It makes it easier for readers to easily parse what's going on.